lib/mpool/src/omf.h

```diff
@@ -143,77 +194,133 @@ OMF_SETGET(struct mblock_metahdr_omf, mh_mcbits, 8);
  * @fh_fileid: file id
  * @fh_rsvd1:  reserved 1
  * @fh_rsvd2:  reserved 2
+ * @fh_crc:    fh crc
  */
 struct mblock_filehdr_omf {
     uint32_t fh_uniq;
     u8       fh_fileid;
     u8       fh_rsvd1;
     uint16_t fh_rsvd2;
+    uint32_t fh_crc;
 } HSE_PACKED;
 
 /* Define set/get methods for mblock_filehdr_omf */
 OMF_SETGET(struct mblock_filehdr_omf, fh_uniq, 32);
 OMF_SETGET(struct mblock_filehdr_omf, fh_fileid, 8);
 OMF_SETGET(struct mblock_filehdr_omf, fh_rsvd1, 8);
 OMF_SETGET(struct mblock_filehdr_omf, fh_rsvd2, 16);
+OMF_SETGET(struct mblock_filehdr_omf, fh_crc, 32);
+
+/**
+ * struct mblock_oid_omf_v1 - per mblock OMF (version 1)
+ */
+struct mblock_oid_omf_v1 {
+    uint64_t mblk_id;
+    uint32_t mblk_wlen;
+    uint32_t mblk_rsvd1;
+    uint64_t mblk_rsvd2;
+} HSE_PACKED;
+
+/* Define set/get methods for mblock_oid_omf_v1 */
+OMF_GET_VER(struct mblock_oid_omf_v1, mblk_id, 64, v1);
+OMF_GET_VER(struct mblock_oid_omf_v1, mblk_wlen, 32, v1);
+OMF_GET_VER(struct mblock_oid_omf_v1, mblk_rsvd1, 32, v1);
+OMF_GET_VER(struct mblock_oid_omf_v1, mblk_rsvd2, 64, v1);
+
+#define MBLOCK_FILE_META_OIDLEN_V1 (sizeof(struct mblock_oid_omf_v1))
 
 /**
  * struct mblock_oid_omf - per mblock OMF
  *
  * @mblk_id:   mblock ID
+ * @mblk_rsvd: reserved
  * @mblk_wlen: mblock write length
- * @mblk_rsvd1: reserved 1
- * @mblk_rsvd2: reserved 2
+ * @mblk_crc:  crc
  */
 struct mblock_oid_omf {
     uint64_t mblk_id;
+    uint64_t mblk_rsvd;
     uint32_t mblk_wlen;
-    uint32_t mblk_rsvd1;
-    uint64_t mblk_rsvd2;
+    uint32_t mblk_crc;
 } HSE_PACKED;
 
 /* Define set/get methods for mblock_oid_omf */
 OMF_SETGET(struct mblock_oid_omf, mblk_id, 64);
+OMF_SETGET(struct mblock_oid_omf, mblk_rsvd, 64);
 OMF_SETGET(struct mblock_oid_omf, mblk_wlen, 32);
-OMF_SETGET(struct mblock_oid_omf, mblk_rsvd1, 32);
-OMF_SETGET(struct mblock_oid_omf, mblk_rsvd2, 64);
+OMF_SETGET(struct mblock_oid_omf, mblk_crc, 32);
 
 #define MBLOCK_FILE_META_OIDLEN (sizeof(struct mblock_oid_omf))
 
+static HSE_ALWAYS_INLINE uint32_t
+omf_mblock_oid_len(uint32_t version)
+{
+    switch (version) {
+    case MBLOCK_METAHDR_VERSION1:
+        return MBLOCK_FILE_META_OIDLEN_V1;
+
+    case MBLOCK_METAHDR_VERSION:
+        return MBLOCK_FILE_META_OIDLEN;
+
+    default:
+        return 0;
```

Comment:
Is this state ever possible? If not, I have just been aborting.

Reply:
Agreed, abort() makes sense here.